src/backupNg/backupNg.spec.js

```diff
@@ -242,4 +232,119 @@ describe("backupNg", () => {
       expect(vmTask.data.id).toBe(vmIdWithoutDisks);
     });
   });
+
+  test("execute three times a rolling snapshot with 2 as retention & revert to an old state", async () => {
+    jest.setTimeout(7e4);
+    const vmId = await xo.createTempVm({
+      name_label: "XO Test Temp",
+      name_description: "Creating a temporary vm",
+      template: config.xoTestTemplateId,
+      VDIs: [
+        {
+          size: 1,
+          SR: config.srs.defaultSr,
+          type: "user",
+        },
+      ],
+    });
+
+    const scheduleTempId = randomId();
+    const { id: jobId } = await xo.createTempBackupNgJob({
+      ...defaultBackupNg,
+      vms: {
+        id: vmId,
+      },
+      schedules: {
+        [scheduleTempId]: DEFAULT_SCHEDULE,
+      },
+      settings: {
+        ...defaultBackupNg.settings,
+        [scheduleTempId]: { snapshotRetention: 2 },
+      },
+    });
+
+    const schedule = await xo.getSchedule({ jobId });
+    expect(typeof schedule).toBe("object");
+    for (let i = 0; i < 3; i++) {
+      const oldSnapshots = xo.objects.all[vmId].snapshots;
+      await xo.call("backupNg.runJob", { id: jobId, schedule: schedule.id });
+      await xo.waitObjectState(vmId, ({ snapshots }) => {
+        // Test on updating snapshots.
+        expect(snapshots).not.toEqual(oldSnapshots);
+      });
+    }
+
+    const { snapshots, videoram: oldVideoram } = xo.objects.all[vmId];
+
+    // Test on the retention, how many snapshots should be saved.
+    expect(snapshots.length).toBe(2);
+
+    const newVideoram = 16;
+    await xo.call("vm.set", { id: vmId, videoram: newVideoram });
+    await xo.waitObjectState(vmId, ({ videoram }) => {
+      expect(videoram).toBe(newVideoram.toString());
+    });
+
+    await xo.call("vm.revert", {
+      snapshot: snapshots[0],
+    });
+
+    await xo.waitObjectState(vmId, ({ videoram }) => {
+      expect(videoram).toBe(oldVideoram);
+    });
+
+    const [
+      {
+        tasks: [
+          {
+            tasks: [task1, task2, task3],
+            ...vmTask
+          },
+        ],
+        ...log
+      },
+    ] = await xo.call("backupNg.getLogs", {
+      jobId,
+      scheduleId: schedule.id,
+    });
+
+    expect(log).toMatchSnapshot({
```

Comment:
Let's thought about a way to automate the logs validation. To not have to update the test each time we add a task in XO.

Reply:
After thought, it will not be easy to do it because it have many specific cases.